lib/child_process.js

```diff
@@ -277,7 +271,7 @@ exports.execFile = function(file /*, args, options, callback*/) {
         if (!encoding)
           _stderr.push(chunk);
         else
-          _stderr += chunk;
+          _stderr += chunk.toString(encoding);
```

Comment:
One alternative would be to keep `setEncoding()`, etc. and only change the length increment lines to something like:

```js
stderrLen += (encoding ? Buffer.byteLength(chunk, encoding) : chunk.byteLength);
```

I haven't benchmarked the difference between that solution and only converting to string at the end though.